docs/tutorials/basic/data.md

```diff
@@ -187,22 +211,361 @@ class SimpleIter:
+The `mx.recordio` package provides a few utility functions
+for such operations, namely: `pack`, `unpack`, `pack_img`, and `unpack_img`.
+
+#### Packing/Unpacking Binary Data.
+
+[__`pack`__](http://mxnet.io/api/python/io.html#mxnet.recordio.pack) and [__`unpack`__](http://mxnet.io/api/python/io.html#mxnet.recordio.unpack) are used for storing float (or 1d array of float) label and binary data:
+
+
+```python
+# pack
+data = 'data'
+label1 = 1.0
+header1 = mx.recordio.IRHeader(flag=0, label=label1, id=1, id2=0)
+s1 = mx.recordio.pack(header1, data)
+
+label2 = [1.0, 2.0, 3.0]
+header2 = mx.recordio.IRHeader(flag=0, label=label2, id=2, id2=0)
+s2 = mx.recordio.pack(header2, data)
+```
+
+
+```python
+# unpack
+print(mx.recordio.unpack(s1))
+print(mx.recordio.unpack(s2))
+```
+
+    (HEADER(flag=0, label=1.0, id=1, id2=0), 'data')
+    (HEADER(flag=3, label=array([ 1.,  2.,  3.], dtype=float32), id=2, id2=0), 'data')
+
+#### Packing/Unpacking Image Data.
+
+MXNet provides [__`pack_img`__](http://mxnet.io/api/python/io.html#mxnet.recordio.pack_img) and [__`unpack_img`__](http://mxnet.io/api/python/io.html#mxnet.recordio.unpack_img) to pack/unpack image data.
+Records packed by `pack_img` can be loaded by `mx.io.ImageRecordIter`.
+
+
+```python
+data = np.ones((3,3,1), dtype=np.uint8)
+label = 1.0
+header = mx.recordio.IRHeader(flag=0, label=label, id=0, id2=0)
+s = mx.recordio.pack_img(header, data, quality=100, img_fmt='.jpg')
+```
+
+```python
+# unpack_img
+print(mx.recordio.unpack_img(s))
+```
+
+    (HEADER(flag=0, label=1.0, id=0, id2=0), array([[1, 1, 1],
+           [1, 1, 1],
+           [1, 1, 1]], dtype=uint8))
+
+
+
+#### Using tools/im2rec.py
+You can also convert raw images into *RecordIO* format using the ``im2rec.py`` utility script that is provided in the MXNet [src/tools]() folder.
+An example how to use the script for converting to *RecordIO* format is shown in the `Image IO` section below.
+
+
+## Image IO
+
+In this section we will learn how to preprocess and load image data in MXNet.
+
+There are 4 ways of loading image data in MXNet.
+   1. Using [__mx.image.imdecode__](http://mxnet.io/api/python/io.html#mxnet.image.imdecode) to load raw image files.
+   2. Using [__`mx.img.ImageIter`__](http://mxnet.io/api/python/io.html#mxnet.image.ImageIter) implemented in Python which is very flexible to customization. It can read from .rec(`RecordIO`) files and raw image files.
+   3. Using [__`mx.io.ImageRecordIter`__](http://mxnet.io/api/python/io.html#mxnet.io.ImageRecordIter) implemented on the MXNet backend in C++. This is less flexible to customization but provides various language bindings.
+   4. Creating a Custom Iterator inheriting `mx.io.DataIter`
+
+
+First set the environment variable `MXNET_HOME` to the root of the MXNet source folder:
+
+
+```python
+# change this to your mxnet location
+MXNET_HOME = '/scratch/mxnet'
+```
+
+
+### Preprocessing Images
```

Comment:
Put this after loading with iterators and explain that you want to use mx.image to write custom iterators when the built-in iterators are not flexible enough.

Also show a very simple custom iterator written with mx.image